src/super_context.ml

```diff
@@ -865,53 +865,120 @@ module PP = struct
           mli) in
     { m with impl ; intf }
 
+  (* laziness is needed to support the case where a user is preprocessing a
+     reason file with action but linting with ppx. *)
+  let lint_module sctx ~(source : Module.t) ~(ast : Module.t Lazy.t) ~dir
+        ~dep_kind ~lint ~lib_name ~scope =
+    let alias = Alias.lint ~dir in
+    match Preprocess_map.find source.name lint with
+    | No_preprocessing -> ()
+    | Action action ->
+      let action = Action.U.Chdir (root_var, action) in
+      Module.iter source ~f:(fun _ (src : Module.File.t) ->
+        let digest_path =
+          Alias.add_action_dep
+            ~action:(Some action)
+            ~action_deps:[Dep_conf.File (String_with_vars.virt __POS__ src.name)]
+            (aliases sctx) alias in
+        let src = Path.relative dir src.name in
+        add_rule sctx
+          (Build.path src
+           >>^ (fun _ -> [src])
+           >>>
+           Build.progn
+             [ Action.run sctx
+                 action
+                 ~dir
+                 ~dep_kind:Required
```

Comment:
Why `Required`?

Reply:
Not sure. What's the difference if it's optional?